src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageDrivenSubscribeTerminator.cs

```diff
@@ -56,7 +56,7 @@ void SendSubscribeMessageWithRetries(string destination, OutgoingMessage subscri
         {
             try
             {
-                dispatcher.Dispatch(subscriptionMessage, new DispatchOptions(destination, new AtLeastOnce(), new List<DeliveryConstraint>(), new ContextBag()));
+                dispatcher.Dispatch(subscriptionMessage, new DispatchOptions(destination, new AtLeastOnce(), new List<DeliveryConstraint>(), new ContextBag())).GetAwaiter().GetResult();
```

Comment:
Did you see the `GetAwaiter().GetResult()`? It will be when the pipeline is async